### DISCUSSION TITLE
Worker monitoring in real time

### DISCUSSION THREAD

**Discussion by PainOchoco** [apalis-dev/apalis#284] [Q&A]
Hello!
I'm trying to find a way to monitor the workers and know when they are working, on what etc.
I saw that apalis-board was a thing but it doesn't seem updated at all. First, do you have plans to bring it back?

Also, when debugging I tried the on_event from the Monitor but it doesn't seem reliable at all; Sometimes it triggers when a worker is started but not reliably, and for the other event types it doesn't triggers at all (Engage, Idle mostly)

Here's a short example that logs with tracing when a worker's status is updated.
```rust
use std::{thread::sleep, time::Duration};

use apalis::prelude::*;
use apalis::redis::RedisStorage;
use serde::{Deserialize, Serialize};
use tracing::{error, info};

#[derive(Debug, Clone, Deserialize, Serialize)]
struct SomeJob {
    id: i32,
}

impl Job for SomeJob {
    const NAME: &'static str = "apalis::SomeJob";
}

async fn print_id(job: SomeJob) {
    sleep(Duration::from_secs(2));
    info!("Job id: {}", job.id);
}

async fn produce_jobs(mut storage: RedisStorage<SomeJob>) {
    for index in 0..20 {
        storage.push(SomeJob { id: index }).await.unwrap();
    }
}

#[tokio::main]
async fn main() {
    std::env::set_var("RUST_LOG", "debug");

    tracing_subscriber::fmt::init();

    let conn = apalis::redis::connect("redis://127.0.0.1/").await.unwrap();
    let storage = RedisStorage::new(conn);
    // This can be in another part of the program
    produce_jobs(storage.clone()).await;

    let worker = WorkerBuilder::new("testing")
        .with_storage(storage.clone())
        .build_fn(print_id);

    Monitor::<TokioExecutor>::new()
        .register_with_count(5, worker)
        .on_event(|e| {
            let worker_id = e.id();
            match e.inner() {
                Event::Start => {
                    info!("Worker [{worker_id}] started");
                }
                Event::Engage => {
                    info!("Worker [{worker_id}] engaged");
                }
                Event::Idle => {
                    info!("Worker [{worker_id}] idle");
                }
                Event::Error(e) => {
                    error!("Worker [{worker_id}] encountered an error: {e}");
                }
                Event::Exit => {
                    info!("Worker [{worker_id}] exited");
                }
                Event::Stop => {
                    info!("Worker [{worker_id}] stopped");
                }
            }
        })
        .run()
        .await
        .unwrap();
}
```

When running this example, the start is not reliable (sometimes I get 1, 2, 3, 4 but not 0 starting).
And when the jobs are getting worked on, no info message gets logged.

Am I doing something wrong?
Thanks for your help!

**Comment by geofmureithi** [comment]
Am working on the board

**Reply by PainOchoco** [reply]
Nice!

**Reply by PainOchoco** [reply]
Please let me know if there's any updates!

**Reply by geofmureithi** [reply]
I will look at the parts not working such as the events triggered, and offer a fix in the next release.
Regarding `apalis-board` initial work has started but its possibly gonna take the rest of the month or more. I do need to work on my job full-time (as open source doesn't really pay 😕) and also need to have some time for personal stuff. It can be challenging but I am trying to allocate time. Hope this makes sense.

**Comment by geofmureithi** [comment]
Looking into those things
